pkg/assets/builder.go

```diff
@@ -160,7 +160,7 @@ func (a *AssetBuilder) RemapImage(image string) (string, error) {
 	}
 
 	if a.AssetsLocation != nil && a.AssetsLocation.ContainerProxy != nil {
-		containerProxy := strings.TrimRight(*a.AssetsLocation.ContainerProxy, "/")
```

Comment:
I'm not sure this is a safe change to make.

If I understand the differences correctly, this would mean that if a ContainerProxy contained a path after the host like `proxy.example.com/path/`, the old behavior would trim it down to `proxy.example.com` but the new behavior will trim it down to `proxy.example.com/path`. I don't know if having a path in ContainerProxy is valid, but if it is this would break existing behavior. Do we know if its valid or not?

Reply:
/cc  @justinsb   for decide 
some works  need to to check this change.

Reply:
/cc @mikesplain

Reply:
I think, with a single character in the second arg, TrimRight will remove multiple occurrences, whereas TrimSuffix will only remove one.  So I think TrimRight is actually probably more correct here (just in case ContainerProxy ends with e.g. `//`), but I don't think it's likely to be a significant issue either way.

Is there a particular reason to make the change, @tanjunchen ?  I agree we haven't been consistent, but I'm not sure which is right. Maybe we need a helper function that we use for all these url-style joins.

Reply:
@rifelpet  @justinsb  Thanks your  review.
Yes, i think it's likely to be a significant issue either way too.
I just found that only `strings.TrimRight` is used here in the kops project, and `strings.TrimSuffix` is used everywhere else. Golang officially recommends this method.

Reply:
@rifelpet  @justinsb  Thanks your  review.
Yes, i think it's likely to be a significant issue either way too.
I just found that only `strings.TrimRight` is used here in the kops project, and `strings.TrimSuffix` is used everywhere else. Golang officially recommends this method.